File: matlab/predict_true.m
function xv= predict_true(xv, V,G, WB,dt)
%
% INPUTS:

xv= [xv(1) + V*dt*cos(G+xv(3,:)); 
     xv(2) + V*dt*sin(G+xv(3,:));
     pi_to_pi(xv(3) + V*dt*sin(G)/WB)];
 

File: cpp/predict_true.cpp
#include "predict_true.h"

void predict_true(VectorXf &xv, float V, float G, int WB, float dt) 
{
	xv(0) = xv(0) + V*dt*cos(G+xv(2));		
	xv(1) = xv(1) + V*dt*sin(G+xv(2));
	vector<float> angle;
	angle.push_back(xv(2) + V*dt*sin(G)/WB); //pi_to_pi takes a vector.		
	pi_to_pi(angle);		
	xv(2) = angle[0];		
}
